### PR TITLE
MAID-1923: refactor/examples: allow examples to compile with 'use-mock-crust' enabled

### PR DESCRIPTION
**Pull request by Fraser999**
The examples will now build with 'use-mock-crust' enabled, but when run will print a warning and
return an error code.  Also, clippy with 'use-mock-crust' is now enabled in the Travis script.

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: examples/ci_test.rs [hunk 1]
```diff
@@ -50,226 +49,235 @@ extern crate unwrap;
...
-        match self.0.kill() {
-            Ok(()) => println!("Killed Node with Process ID #{}", self.0.id()),
-            Err(err) => {
-                println!("Error killing Node with Process ID #{} - {:?}",
-                         self.0.id(),
-                         err)
+#[cfg(feature = "use-mock-crust")]
+fn main() {
+    println!("This example should be built without `--features=use-mock-crust`.");
+    ::std::process::exit(78);
```
Comment (maqi) [inline examples/ci_test.rs]: shall we have a comment explaining why 78 is chosen?

Comment (Fraser999) [inline examples/ci_test.rs]: I didn't think it was important, but I can do :)

Comment (Fraser999) [inline examples/ci_test.rs]: Done now.

**Pull request merged**